06-building-a-program-start-to-finish/writing-tests/output/spec/InvertedIndexSpec.js

```diff
@@ -0,0 +1,277 @@
+      const result = ['Alice in Wonderland',
+        'The Lord of the Rings: The Fellowship of the Ring.',
+        'The Lord of the Rings.'];
+      expect(InvertedIndex.getTitles(testFile2)).toEqual(result);
+    });
+  });
+
+  describe('Tests for the createIndex method', () => {
+    it('should return error message for non-Array objects', () => {
+      expect(() => { newIndex.createIndex('hello', 'Hello World'); })
+          .toThrow(new Error(`This file's structure is invalid.
+       Only array of objects are allowed.`));
+      expect(() => { newIndex.createIndex('test', new Set(['a', 'the'])); })
+          .toThrow(new Error(`This file's structure is invalid.
+       Only array of objects are allowed.`));
+      expect(() => { newIndex.createIndex('test', 200); })
+          .toThrow(new Error(`This file's structure is invalid.
+       Only array of objects are allowed.`));
+    });
+
+    it('should return error message for files with improper key values', () => {
+      const test = [
+        {
+          tite: 'Alice in Wonderland',
+          text: 'Alice. falls- into@ a+ rabbit hole.',
+        },
+      ];
+
+      expect(() => { newIndex.createIndex('doc1.json', invalidTestFile); })
+        .toThrow(new Error(`Your JSON file does not have the expected format.
+          Documents are to have title and text keys alone.`));
+      expect(() => { newIndex.createIndex('test.json', test); })
+        .toThrow(new Error(`Your JSON file does not have the expected format.
+          Documents are to have title and text keys alone.`));
+    });
+
+    it('should return the correct indices for words in documents', () => {
+      const result = { a: [0],
+        alice: [0],
+        alliance: [1],
+        an: [1],
+        elf: [1],
+        falls: [0],
+        hole: [0],
+        into: [0],
+        man: [1],
+        of: [1],
+        rabbit: [0],
+        unusual: [1] };
+      expect(newIndex.createIndex('doc.json', shortFile2)['doc.json'][0])
+      .toEqual(result);
+    });
+
+    it(`returns an object of key name and value of array of file indices,
+     file name and file titles`, () => {
+      const result = [{ a: [0],
+        alice: [0],
+        alliance: [1],
+        an: [1],
+        elf: [1],
+        falls: [0],
+        hole: [0],
+        into: [0],
+        man: [1],
+        of: [1],
+        rabbit: [0],
+        unusual: [1] }, 'doc.json', ['Alice in Wonderland',
+          'The Lord of the Rings: The Fellowship of the Ring.']];
+      expect(newIndex.createIndex('doc.json', shortFile2)['doc.json'])
+      .toEqual(result);
+    });
+
+    it('should only accept an array of objects', () => {
+      const testObj = {
+        title: 'Alice in Wonderland',
+        text: 'Alice. falls- into@ a+ rabbit hole.',
+      };
+
+      expect(() => { newIndex.createIndex('test.json', testObj); })
+      .toThrow(new Error(`This file's structure is invalid.
+       Only array of objects are allowed.`));
+    });
+  });
```

Comment:
Most specifications in your createIndex() seem more like validation test which I believe should be part of your validateFile();